ivy/array/set.py

```diff
@@ -1,20 +1,17 @@
+# for review
 # global
 import abc
-from typing import Tuple
+from typing import Optional, Tuple
 
-# local
-import ivy
 
-# ToDo: implement all methods here as public instance methods
+import ivy
 
 
 class ArrayWithSet(abc.ABC):
-    def unique_counts(
-        self: ivy.Array
-    ) -> Tuple[ivy.Array, ivy.Array]:
+    def unique_counts(self: ivy.Array) -> Tuple[ivy.Array, ivy.Array]:
```

Comment:
looking at the example, seems as though this is actually a `NamedTuple`, does this exist in the typing module?

Reply:
yes `NamedTuple` exists and it seems to be more suitable here, have changed it